src/App.tsx

```diff
@@ -1,26 +1,93 @@
-import React from "react";
-import logo from "./logo.svg";
+import { useState } from "react";
 import "./App.css";
+import { InputTask } from "./components/listItem";
+import { MyButton } from "./components/myButton";
+import { MySearching } from "./components/MySearching";
 
 function App() {
+  const [tasks, setTasks] = useState<{ isDone: boolean; title: string }[]>([]);
+  const [inputValue, setInputValue] = useState<string>("");
+  const [searchValue, setSearchValue] = useState<string>("");
+
+  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setInputValue(e.target.value);
+  };
+
+  const handleCreateTask = () => {
+    if (inputValue.trim() === "") return;
+    const newTask = { title: inputValue, isDone: false };
+    setTasks((prev) => [...prev, newTask]);
+    setInputValue("");
+  };
+  const handleKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
+    if (e.key === "Enter") {
+      handleCreateTask();
+    }
+  };
+  const handleDoneClick = (index: number) => {
+    setTasks((prevTasks) =>
+      prevTasks.map((task, i) =>
+        i === index ? { ...task, isDone: true } : task,
+      ),
+    );
+  };
+  const handleDelClick = (index: number)=>{
+    setTasks((prevTasks) => prevTasks.filter((_, i) => i !== index)
+    );
+  };
+  const handleSearching = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setSearchValue(e.target.value);
+  };
+  const filteredTasks = tasks.filter((task) =>
```

Comment:
this function will be fired on every rerender, use `useCallback` hook with `searchValue` dependency inside or wrap this in a function and fire it on `onInput` event